clang/lib/3C/CheckedRegions.cpp

```diff
@@ -154,13 +159,26 @@ bool CheckedRegionFinder::VisitDoStmt(DoStmt *S) {
 
 bool CheckedRegionFinder::VisitCompoundStmt(CompoundStmt *S) {
   // Visit all subblocks, find all unchecked types
-  bool Localwild = 0;
+  bool Localwild = false;
   for (const auto &SubStmt : S->children()) {
     CheckedRegionFinder Sub(Context, Writer, Info, Seen, Map, EmitWarnings);
     Sub.TraverseStmt(SubStmt);
     Localwild |= Sub.Wild;
   }
 
+  // If we are a function def, need to check return type
+  if (isTopLevel(Context, S)) {
```

Comment:
Can you explain what situation this is covering that the old code, which didn't check whether you were in the "top level" in such a general way, missed? 

Also, I notice that `isTopLevel` is silently casting a pointer to a `bool`, is that right? Seems icky. Why not instead return `!Parents.empty()` ?